server/src/main/java/com/cloud/network/lb/LoadBalancingRulesManagerImpl.java

```diff
@@ -340,7 +341,27 @@ private LbAutoScaleVmGroup getLbAutoScaleVmGroup(AutoScaleVmGroupVO vmGroup, Aut
         String csUrl = ApiServiceConfiguration.ApiServletPath.value();
         Network.Provider provider = getLoadBalancerServiceProvider(lb);
         if (Network.Provider.Netscaler.equals(provider)) {
-            autoScaleManager.checkAutoScaleUser(autoScaleVmProfile.getAutoScaleUserId(), vmGroup.getAccountId());
+            Long autoscaleUserId = autoScaleVmProfile.getAutoScaleUserId();
+            if (autoscaleUserId == null) {
+                throw new InvalidParameterValueException("autoscaleUserId is required but not specified");
+            }
+            User user = _userDao.findById(autoscaleUserId);
+            if (user == null) {
+                throw new InvalidParameterValueException("Unable to find user by id " + autoscaleUserId);
+            }
+            apiKey = user.getApiKey();
+            secretKey = user.getSecretKey();
+            if (apiKey == null) {
+                throw new InvalidParameterValueException("apiKey for user: " + user.getUsername() + " is empty. Please generate it");
+            }
+
+            if (secretKey == null) {
+                throw new InvalidParameterValueException("secretKey for user: " + user.getUsername() + " is empty. Please generate it");
+            }
+
+            if (csUrl == null || csUrl.contains("localhost")) {
+                throw new InvalidParameterValueException(String.format("Global setting %s has to be set to the Management Server's API end point", ApiServiceConfiguration.ApiServletPath.key()));
+            }
```

Comment:
should this than not be the new contents of `checkAutoScaleUser()`?

Reply:
@DaanHoogland 
You might notice that I have changes to use `checkAutoScaleUser` and then reverted back.
because `apiKey` and `secretKey` are updated in these lines, but `checkAutoScaleUser` will not update them.